shared-libs/transitions/src/schedule/due_tasks.js

```diff
@@ -1,4 +1,4 @@
-// TODO: this doesn't need to exist. We can just work this out dynamically when
+// ToThinkAbout: this doesn't need to exist. We can just work this out dynamically when
```

Comment:
Sonar complains about this.

Reply:
As it should! TODOs should be GH issues...

Reply:
I personally don't agree with this todo :D

Reply:
Feel free to delete it if it's wrong!

Reply:
I'm not sure if it's wrong. TBH we don't _actually_ keep track of how many messages get processed by one of these runs, and what impact this would have if the `api/sms` endpoint would wait for this processing. 

What would we do in API? Run the due_dates optimistically and only do one batch at a time, and optimistically send from whatever messages we have pending? Run multiple batches until we have at lease one full load (25) pending messages or we have cleared the scheduled queue? 

What if someone "spams"  `api/sms`? Spamming is not even required, it would be enough to have concurrent requests that end up firing updates for the same set of docs and just failing with conflicts? 

I already believe that `api/sms` is bugged to update message states _before_ actually confirming that messages have been sent to gateway, which makes us need to query for two statuses (pending and forwarded-to) to get pending messages.